### PR TITLE
fix: [CDS-40941]: upgraded to v2 variables api

### PR DESCRIPTION
**Pull request by samarth-sharma31**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by samarth-sharma31** [comment]
retrigger titlecheck

**Comment by samarth-sharma31** [comment]
dockerize

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Passed_** 
 


            
 #### 2 Issues 
* Code issue 2 Code issues 

 [Details](https://app.embold.io/organization/gh/harness/repositories/e864badd0b95ab81c060b15b85f15701/pull-request/10271)

**Comment by samarth-sharma31** [comment]
retrigger jest

**Comment by samarth-sharma31** [comment]
trigger cypress

File: src/modules/70-pipeline/components/PipelineStudio/PiplineHooks/useVariablesExpression.ts [hunk 1]
```diff
@@ -101,35 +120,26 @@ export function useVariablesExpression(): { expressions: string[] } {
...
     if (!templateInitLoading && !isEmpty(templateMetadataMap)) {
-      const expression = pickExpressionsFromMetadataMap(templateMetadataMap, localStageKeys, 'yamlProperties')
-      const outputExpression = pickExpressionsFromMetadataMap(
-        templateMetadataMap,
-        localStageKeys,
-        'yamlOutputProperties'
-      )
-      setExpressions([...expression, ...outputExpression])
+      const { expressionsList, outputExpressions, extraExpressions, extraOutputExpressions } =
+        getAllExpressionsFromMetadataMap({ metadataMap: templateMetadataMap, localStageKeys })
```
Comment (udayvunnam) [inline src/modules/70-pipeline/components/PipelineStudio/PiplineHooks/useVariablesExpression.ts:140]: we may not need an effect here, These can be extracted in regular flow

**Review by udayvunnam — APPROVED** [review]

**Review by chetanvekariya-harness — DISMISSED** [review]

File: src/modules/70-pipeline/components/PipelineStudio/PiplineHooks/useVariablesExpression.ts (hunk 1 above)
Comment (samarth-sharma31) [reply src/modules/70-pipeline/components/PipelineStudio/PiplineHooks/useVariablesExpression.ts]: Okay will check this and remove it if not required in a followup PR

**Review by yc147 — DISMISSED** [review]

**Review by chetanvekariya-harness — APPROVED** [review]

**Review by yc147 — APPROVED** [review]

**Review by kapilkumawat86 — APPROVED** [review]

**Pull request merged**